katello/hooks/pre_commit/14-cdn_setting.rb

```diff
@@ -0,0 +1,12 @@
+if module_enabled?('katello')
+  # Using @kafo.config.answers because it will have the answers prior to being
+  # committed to disk and being overwritten
+  # https://github.com/theforeman/foreman-installer/pull/477#pullrequestreview-370785117
```

Comment:
I think this is technically not entirely correct. `param` (and thus `param_value`) use the same data structure (`kafo.config.answers`). However, `param` goes through `Kafo::PuppetModule` which uses the actual Puppet module (via a cache in production) to see which params actually exist. Since it doesn't exist, `param('katello', 'cdn_ssl_version')` returns nil.

Would something like this be better?

```suggestion
  # Using @kafo.config.answers because param() goes through PuppetModule. At
  # this point the parameter doesn't exist in the module on disk so param()
  # returns nil.
```

Reply:
I find this confusing personally. But I find explaining all of it confusing to be honest. As there are nuances depending on where you are at in the process as to what each represents. At this stage:

 * kafo.config.answers represents the old parameters and values on disk
 * param represents the current state of params before the transaction commits the new values to disk

Reply:
You're right, but in the current committed comment I'm missing explicitly that `param()` doesn't work. The reason I ask for this is that it might be very tempting to think about refactoring it using built in hooks, forgetting that it doesn't work.